ocaml/xapi/xapi_pci_helpers.ml

```diff
@@ -172,3 +172,36 @@ let get_host_pcis () =
 let igd_is_whitelisted ~__context pci =
   let vendor_id = Db.PCI.get_vendor_id ~__context ~self:pci in
   List.mem vendor_id !Xapi_globs.igd_passthru_vendor_whitelist
+
+let update_dom0_access ~__context ~pci ~action =
+  let db_new =
+    ref
+      ( match action with
+      | `enable ->
+          Pciops.unhide_pci ~__context pci ;
+          `enabled
+      | `disable ->
+          Pciops.hide_pci ~__context pci ;
+          `disabled
+      )
+  in
+  let expr = Printf.sprintf "field \"PCI\"=\"%s\"" (Ref.string_of pci) in
+  let pgpus = Db.PGPU.get_all_records_where ~__context ~expr in
+  List.iter
+    (fun (pgpu_ref, _) ->
+      let db_current = Db.PGPU.get_dom0_access ~__context ~self:pgpu_ref in
```

Comment:
Again: would not put the focus on `db` here but on something like `state`, `access`, or `visibility`.

Reply:
`new_access` and `current_access`?